codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -156,14 +175,10 @@ public void sendFlowControlled(ChannelHandlerContext ctx, Http2Stream stream,
         }
         // Save the context. We'll use this later when we write pending bytes.
         this.ctx = ctx;
-        try {
-            FlowState state = state(stream);
-            state.newFrame(payload);
-            state.writeBytes(state.writableWindow());
-            flush();
-        } catch (Throwable e) {
-            payload.error(e);
-        }
+        FlowState state = state(stream);
+        state.newFrame(payload);
+        state.writeBytes(state.writableWindow());
+        flush();
```

Comment:
> so we basically would have to put everything under try catch except for the writesBytes call, because we must not call error after writeComplete or error has been called.

This is what I had in mind.  Lets see how ugly this makes things :)